Dockerfile.api

```diff
@@ -1,60 +1,45 @@
 ARG PORT
 
 ################ base ################
-FROM python:3.10 AS base
+FROM ghcr.io/astral-sh/uv:python3.11-bookworm AS base
 
 ################ builder ################
 FROM base AS builder
 
-# install poetry in its own environment
-ENV POETRY_VERSION=2.1.3 \
-    POETRY_HOME=/opt/poetry
-RUN python -m venv $POETRY_HOME && \
-    $POETRY_HOME/bin/pip install -U pip poetry==$POETRY_VERSION
+WORKDIR /home/nonroot
+COPY stats-entities ./stats-entities
+COPY stats-api/stats_api ./stats-api/stats_api
+COPY stats-api/pyproject.toml stats-api/uv.lock ./stats-api
 
-ENV PATH="/opt/poetry/bin:$PATH"
-
-# install dependencies in their own environment
-WORKDIR /home
-COPY poetry.lock pyproject.toml ./
-COPY stats ./stats/
-RUN poetry config virtualenvs.in-project true && \
-    poetry install --no-interaction --no-ansi \
-    --without test
+WORKDIR /home/nonroot/stats-api
+RUN uv sync --locked --no-dev
 
 ################ runtime ################
-FROM base AS runtime
-COPY --from=builder /opt/poetry /opt/poetry
-COPY --from=builder /home /home
+FROM builder AS runtime
+
+ENV PATH="/home/nonroot/stats-api/.venv/bin:$PATH"
 
-ENV PATH="/opt/poetry/bin:$PATH"
+# run as non-root user for security reasons
+RUN useradd nonroot
+RUN chmod -R u+x /home/nonroot
```

Comment:
/home/nonroot will be owned by `root` I guess, because it was created in `base` like this.
Is there any chance that the python process needs to write files? (What about .pyc files?)